### PR TITLE
#5810 Fixed lsmagic output.

### PR DESCRIPTION
**Pull request by michalgce**

### CODE REVIEW

File: kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/item/MagicCommandType.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.twosigma.beakerx.kernel.commands.item;
```
Comment (scottdraves) [inline kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/item/MagicCommandType.java:17]: missing banner

File: kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/item/MagicCommandType.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  private final String parameters;
+  private final MagicCommandFunctionality functionality;
+  private final String suitableKernelsNames;
+
+  public MagicCommandType(String command, String parameters,
+      MagicCommandFunctionality functionality, String suitableKernelsNames) {
+    this.command = command;
+    this.parameters = parameters;
+    this.functionality = functionality;
+    this.suitableKernelsNames = suitableKernelsNames;
```
Comment (scottdraves) [inline kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/item/MagicCommandType.java]: this isn't the right way to do this.
instead, have the kernel add the commands.
base kernel adds magics that work in all kernels.

File: kernel/base/src/main/java/com/twosigma/beakerx/kernel/KernelFunctionality.java [hunk 3]
```diff
@@ -75,6 +75,8 @@
 
   void removeImport(ImportPath anImport);
 
+  String getName();
```
Comment (scottdraves) [inline kernel/base/src/main/java/com/twosigma/beakerx/kernel/KernelFunctionality.java]: this method is not needed, see below.

File: kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/item/MagicCommandType.java (hunk 2 above)
Comment (michalgce) [reply kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/item/MagicCommandType.java]: good idea!

**Pull request merged**